lib/common/domain/router/routes.dart

```diff
@@ -58,8 +60,16 @@ RouteBase _statefulShellRoute({
           routes: [
             GoRoute(
               path: PopularMoviesPage.routeName,
-              builder: (context, state) => PopularMoviesPage(),
+              builder: (context, state) => const PopularMoviesPage(),
               routes: [
+                GoRoute(
+                  path: MovieDetailsPage.routeName.removeLeadingSlash,
+                  parentNavigatorKey: rootNavigatorKey,
+                  builder: (context, state) {
+                    final movie = state.extra as Movie;
```

Comment:
Ovo je ok ako se targetira samo mobilna aplikacija, da želiš da stvar radi i na webu, onda ne radi ako bi išao s liste na detalje, back na listu i onda u browseru ti ostane forward gumb od browsera, kad klikneš na to onda pukne baš ovdje. Isto bi bilo da napraviš browser refresh na MovieDetailsPage ruti. Da želiš to složiti web-friendly onda bi predavao movie id u samoj ruti i onda ovdje čitao id i onda u notifieru od movie details pagea dohvaćao detalje filma s api endpointa za detalje.